lib/android.dart

```diff
@@ -290,30 +256,31 @@ List<String> transformAndroidManifestWithNewLauncherIcon(
       // repeat as often as wanted with no quote at start: [^"]*(\"[^"]*)*
       // escaping the slash to place in string: [^"]*(\\"[^"]*)*"
       // result: any string which does only include escaped quotes
-      return line.replaceAll(RegExp(r'android:icon="[^"]*(\\"[^"]*)*"'),
-          'android:icon="@mipmap/$iconName"');
+      return line.replaceAll(RegExp(r'android:icon="[^"]*(\\"[^"]*)*"'), 'android:icon="@mipmap/$iconName"');
     } else {
       return line;
     }
   }).toList();
 }
 
-/// Retrieves the minSdk value from the Android build.gradle file or local.properties file
-int minSdk() {
+/// Retrieves the minSdk value from the
+/// - flutter.gradle: `'$FLUTTER_ROOT/packages/flutter_tools/gradle/flutter.gradle'`
+/// - build.gradle: `'android/app/build.gradle'`
+/// - local.properties: `'android/local.properties'`
+///
+/// If found none returns 0
+int? minSdk() {
   final androidGradleFile = File(constants.androidGradleFile);
   final androidLocalPropertiesFile = File(constants.androidLocalPropertiesFile);
 
-  // look in build.gradle first
-  final minSdkValue = getMinSdkFromFile(androidGradleFile);
-
-  // look in local.properties. Didn't find minSdk, assume the worst
-  return minSdkValue != 0
-      ? minSdkValue
-      : getMinSdkFromFile(androidLocalPropertiesFile);
+  // looks for minSdk value in build.gradle, flutter.gradle & local.properties.
+  return getMinSdkFlutterGradle(androidLocalPropertiesFile) ??
+      getMinSdkFromFile(androidGradleFile) ??
+      getMinSdkFromFile(androidLocalPropertiesFile);
 }
 
 /// Retrieves the minSdk value from [File]
-int getMinSdkFromFile(File file) {
+int? getMinSdkFromFile(File file) {
```

Comment:
Since we aren't using this outside of this file, we can make this method private

```suggestion
int? _getMinSdkFromFile(File file) {
```

Reply:
fixed in 7e3b5a4